pytest_bdd/scenario.py

```diff
@@ -111,8 +110,12 @@ def _execute_step_function(request, scenario, step, step_func):
         kw["step_func_args"] = kwargs
 
         request.config.hook.pytest_bdd_before_step_call(**kw)
+        target_fixture = getattr(step_func, "target_fixture", None)
         # Execute the step.
-        step_func(**kwargs)
+        return_value = step_func(**kwargs)
+        if target_fixture:
+            inject_fixture(request, target_fixture, return_value)
```

Comment:
@olegpidsadnyi the `getfixturevalue` is returning generator instead of a value since the `return_value` is a generator for "yielding" fixture.

```python
@when("a cleanup is necessary", target_fixture="foo")
def foo(request):
    obj = setup()
    yield obj.work()
    obj.clean()
```